### PR TITLE
add product to cart

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/76017

## WHAT (optional)
- Add product to cart

## HOW
- When a user clicks on the 'Add to Cart' button, if they haven't logged in yet, an error message is shown. If the user is logged in, the product is added to the cart, and a success message is displayed.

## Evidence (Screenshot or Video)
image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

**Comment by ghost** [comment]
ready

File: src/controllers/cart.controller.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  public addToCart = asyncHandler(
+    async (req: Request, res: Response) => {
+      const user = (req.session as CustomSessionData).user;
+      const productId = Number(req.body?.productId);
+      if(user!==undefined){
+        try {
+          await this.cartService.addToCart(user, productId);
+          res.status(200).send({ 
+            success: true, 
+            message: 'Product added to cart successfully.',
```
Comment (huongnt-2545) [inline src/controllers/cart.controller.ts]: I18n message

File: src/services/cart.service.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    this.userRepository = AppDataSource.getRepository(UserEntity);
+    this.productRepository = AppDataSource.getRepository(ProductEntity);
+  }
+
+  public async addToCart(
+    user: UserEntity, 
+    productId: number, 
+  ): Promise< string | CartProductEntity> {
+    const product = await this.productRepository
+    .findOne({where: {id: productId}});
```
Comment (huongnt-2545) [inline src/services/cart.service.ts]: tách xử lý tìm kiếm product qua service của product e nha

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**